generator/dart/generated/google_cloud_functions_v2/lib/cloudfunctions.dart

```diff
@@ -327,33 +327,33 @@ class Function$ extends Message {
 }
 
 /// Describes the current state of the function.
-class Function$$State extends Enum {
+class Function$State extends Enum {
   /// Not specified. Invalid state.
-  static const stateUnspecified = Function$$State('STATE_UNSPECIFIED');
+  static const stateUnspecified = Function$State('STATE_UNSPECIFIED');
 
   /// Function has been successfully deployed and is serving.
-  static const active = Function$$State('ACTIVE');
+  static const active = Function$State('ACTIVE');
 
   /// Function deployment failed and the function is not serving.
-  static const failed = Function$$State('FAILED');
+  static const failed = Function$State('FAILED');
 
   /// Function is being created or updated.
-  static const deploying = Function$$State('DEPLOYING');
+  static const deploying = Function$State('DEPLOYING');
 
   /// Function is being deleted.
-  static const deleting = Function$$State('DELETING');
+  static const deleting = Function$State('DELETING');
 
   /// Function deployment failed and the function serving state is undefined.
   /// The function should be updated or deleted to move it out of this state.
-  static const unknown = Function$$State('UNKNOWN');
+  static const unknown = Function$State('UNKNOWN');
 
-  const Function$$State(super.value);
+  const Function$State(super.value);
```

Comment:
If this is intended as an enum, consider either making it an `enum`, or at least making the constructor private, so no more instances will be created.
Then you can use `identical` in `operator==`.
(But if you have to accept any string value in `fromJson`, you can't do that. Must you accept any string value?)

Reply:
Yeah, I think one requirement is that the set of enum values we have to deal with isn't closed. We may get an enum value over the wire that we didn't know about at generation time. We'd need to:

- not throw an exception
- possibly allow the user to still reason about that unexpected value
- probably be able to round-trip the same enum value back to the server (if contained in some parent object)